tests/test_analog.py

```diff
@@ -148,6 +148,11 @@ def test_hamiltonianevolution_with_types(
     assert result.size() == (batch_size,)
     assert torch.allclose(result, target, rtol=RTOL, atol=ATOL)
 
+    # test cached
```

Comment:
Wouldn't it be possible to check for numbers or equivalence of some sort ? Just checking for the length of an object is a bit meaningless.